File: benchmark/results/python-multilang/run-004/b-baseline/output/metaengine_demo/shipping/services/carrier_gateway.py
"""CarrierGateway for the shipping domain."""
from typing import Optional

from metaengine_demo.shipping.aggregates.shipment import Shipment


class CarrierGateway:
    """Gateway integrating with external carriers."""

    def create(self, input: Optional[Shipment]) -> Shipment:
        """Hand off a new Shipment to a carrier."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Shipment]:
        """Look up a Shipment at the carrier by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Shipment]:
        """List Shipments at the carrier up to limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Cancel a Shipment at the carrier by id."""
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-monolith-opus-inline/run-001/b-baseline/output/src/orders/fulfillment/services/CarrierGateway.ts
import { Id } from "../../../shared/value-objects/Id";
import { Shipment } from "../aggregates/Shipment";

/** Gateway to external shipping carriers. */
export class CarrierGateway {
  create(input: Partial<Shipment>): Shipment {
    void input;
    throw new Error("not implemented");
  }

  findById(id: Id): Shipment | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Shipment[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: Id): void {
    void id;
    throw new Error("not implemented");
  }
}
